tests/phpunit/integration/Modules/Analytics/AdvancedTrackingTest.php

```diff
@@ -0,0 +1,190 @@
+ *
+ * @package   Google\Site_Kit\Tests\Modules\Analytics
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Modules\Analytics;
+
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Plugin_Detector;
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking;
+use Google\Site_Kit\Tests\Modules\MockPluginDetector;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Code_Injector;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\Measurement_Event_Pipe;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\CF7_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\FormidableForms_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\NinjaForms_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\Woocommerce_Event_List;
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Measurement_Events\WPForms_Event_List;
+
+/**
+ * Class AdvancedTrackingTest
+ * @package Google\Site_Kit\Tests\Modules\Analytics
+ */
+class AdvancedTrackingTest extends TestCase {
+
+	private $supported_plugins;
+
+	private $mock_plugin_detector;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->supported_plugins    = array(
+			'Contact Form 7'   => array(
+				'check_name'        => 'WPCF7_PLUGIN_DIR',
+				'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+				'event_config_list' => new CF7_Event_List(),
+			),
+			'Formidable Forms' => array(
+				'check_name'        => 'load_formidable_forms',
+				'check_type'        => Plugin_Detector::TYPE_FUNCTION,
+				'event_config_list' => new FormidableForms_Event_List(),
+			),
+			'Ninja Forms'      => array(
+				'check_name'        => 'NF_PLUGIN_DIR',
+				'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+				'event_config_list' => new NinjaForms_Event_List(),
+			),
+			'WooCommerce'      => array(
+				'check_name'        => 'WC_PLUGIN_FILE',
+				'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+				'event_config_list' => new Woocommerce_Event_List(),
+			),
+			'WPForms'          => array(
+				'check_name'        => 'WPFORMS_PLUGIN_DIR',
+				'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+				'event_config_list' => new WPForms_Event_List(),
+			),
+			'WPForms Lite'     => array(
+				'check_name'        => 'WPFORMS_PLUGIN_DIR',
+				'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+				'event_config_list' => new WPForms_Event_List(),
+			),
+		);
+		$this->mock_plugin_detector = new MockPluginDetector();
+	}
+
+	/**
+	 * Tests if the sets of events in the active plugins are included in the tracking event configurations.
+	 */
+	public function test_event_configurations() {
```

Comment:
Let's call all test methods based on the function/method they test. Additional suffixes are okay, but we should follow the pattern `test_{method_name}[_optional_suffixes_to_briefly_describe_the_test_if_there_are_multiple_for_the_same_method]` :)

Reply:
This issue has been resolved.